spec/models/inventory_spec.rb

```diff
@@ -0,0 +1,173 @@
+      inventory.model_id = nil
+      expect(inventory).to_not be_valid
+    end
+    it 'is not valid without memory_size' do
+      inventory.memory_size = nil
+      expect(inventory).to_not be_valid
+    end
+    it 'is not valid without manufactoring_year' do
+      inventory.manufactoring_year = nil
+      expect(inventory).to_not be_valid
+    end
+    it 'is not valid without os_version' do
+      inventory.os_version = nil
+      expect(inventory).to_not be_valid
+    end
+    it 'is not valid without color' do
+      inventory.color = nil
+      expect(inventory).to_not be_valid
+    end
+    it 'is not valid without price' do
+      inventory.price = nil
+      expect(inventory).to_not be_valid
+    end
+    it 'is not valid if model is not existed' do
+      inventory.model_id = 5
+      expect(inventory).to_not be_valid
+    end
+    it 'is not valid if memory_size is not a positive integer number' do
+      inventory.memory_size = -45
+      expect(inventory).to_not be_valid
+      inventory.memory_size = 18.5
+      expect(inventory).to_not be_valid
+    end
+    it 'is not valid if manufactoring_year is not a reasonable year (positive integer and <= current year)' do
+      inventory.manufactoring_year = -45
+      expect(inventory).to_not be_valid
+      inventory.manufactoring_year = 25.5
+      expect(inventory).to_not be_valid
+      inventory.manufactoring_year = Time.zone.now.year + 1
+      expect(inventory).to_not be_valid
+    end
+    it 'is not valid if price is negative price' do
+      inventory.price = -1
+      expect(inventory).to_not be_valid
+    end
+  end
+
+  describe '.scope' do
+    before(:all) do
+      seed_data
+    end
+
+    context 'records found' do
+      it 'by brand' do
+        brand = Brand.find_by(name: 'Apple')
+        inventories = Inventory.by_brand_id(brand.id)
+        expect(inventories.count).to eq(3)
+      end
+      it 'by model' do
+        model = Model.find_by(name: 'iPhone X')
+        inventories = Inventory.by_model_id(model.id)
+        expect(inventories.count).to eq(2)
+      end
+      it 'by manufactoring_year lower bound' do
+        inventories = Inventory.by_manufactoring_year_lower(2020)
+        expect(inventories.count).to eq(3)
+      end
+      it 'by manufactoring_year upper bound' do
+        inventories = Inventory.by_manufactoring_year_upper(2019)
+        expect(inventories.count).to eq(2)
+      end
+      it 'by memory_size range' do
+        inventories = Inventory.by_memory_size('Under 16GB')
+        expect(inventories.count).to eq(1)
+        inventories = Inventory.by_memory_size('From 16GB to 64GB')
+        expect(inventories.count).to eq(1)
+        inventories = Inventory.by_memory_size('From 64GB to 256GB')
+        expect(inventories.count).to eq(2)
+        inventories = Inventory.by_memory_size('Over 256GB')
+        expect(inventories.count).to eq(2)
+      end
+      it 'by price range' do
```

Comment:
một it nên viết 1 case thôi, nếu nhiều case khác nhau thì chia ra 1 2 it, viết để thấy được sự khác biệt.